pkg/cvo/internal/operatorstatus.go

```diff
@@ -210,31 +216,44 @@ func waitForOperatorStatusToBeDone(ctx context.Context, interval time.Duration,
 			}
 		}
 
+		nestedMessage := fmt.Errorf("cluster operator %s conditions: available=%v, progressing=%v, degraded=%v",
+			actual.Name, available, progressing, degraded)
+
+		if !available {
+			lastErr = &payload.UpdateError{
+				Nested:       nestedMessage,
+				UpdateEffect: payload.UpdateEffectFail,
+				Reason:       "ClusterOperatorNotAvailable",
+				Message:      fmt.Sprintf("Cluster operator %s is not available", actual.Name),
+				Name:         actual.Name,
+			}
+			return false, nil
+		}
+
 		condition := failingCondition
 		if degradedCondition != nil {
 			condition = degradedCondition
 		}
 		if condition != nil && condition.Status == configv1.ConditionTrue {
-			message := fmt.Sprintf("Cluster operator %s is reporting a failure", actual.Name)
 			if len(condition.Message) > 0 {
-				message = fmt.Sprintf("Cluster operator %s is reporting a failure: %s", actual.Name, condition.Message)
+				nestedMessage = fmt.Errorf("cluster operator %s is reporting a message: %s", actual.Name, condition.Message)
 			}
 			lastErr = &payload.UpdateError{
-				Nested:  errors.New(lowerFirst(message)),
-				Reason:  "ClusterOperatorDegraded",
-				Message: message,
-				Name:    actual.Name,
+				Nested:       nestedMessage,
+				UpdateEffect: payload.UpdateEffectFailAfterInterval,
+				Reason:       "ClusterOperatorDegraded",
+				Message:      fmt.Sprintf("Cluster operator %s is degraded", actual.Name),
+				Name:         actual.Name,
 			}
 			return false, nil
 		}
 
 		lastErr = &payload.UpdateError{
-			Nested: fmt.Errorf("cluster operator %s is not done; it is available=%v, progressing=%v, degraded=%v",
-				actual.Name, available, progressing, degraded,
-			),
-			Reason:  "ClusterOperatorNotAvailable",
-			Message: fmt.Sprintf("Cluster operator %s has not yet reported success", actual.Name),
-			Name:    actual.Name,
+			Nested:       nestedMessage,
+			UpdateEffect: payload.UpdateEffectNone,
+			Reason:       "ClusterOperatorNotAvailable",
```

Comment:
We might want a different reason for this, but we can twiddle that in follow-up work.